front/pages/api/w/[wId]/data_sources/[name]/databases/csv.ts

```diff
@@ -261,6 +265,17 @@ async function rowsFromCsv(
     // We keep the parsed values from the first parser that succeeds for all non-null values in the column.
     parsedValuesByCol[col] = (() => {
       for (const parser of [
+        // date/datetime
+        (v: string) => {
+          const date = new Date(v);
```

Comment:
in modern JS, all of these work:
```
new Date("2023-12-01T15:43:00Z")
new Date("2023-12-01T15:43:00+00:00")
new Date("2023-12-01T15:43:00-05:00")
new Date("2023-12-01T15:43:00.000Z")
new Date("2023-12-01T15:43:00.000+00:00")
new Date("2023-12-01T20:43:00.000-05:00")
new Date("2023-12-01")
new Date("2023-12")
new Date("2023")
new Date("2023T15:43:00Z")
new Date("2023-12-01 15:43:00Z")
new Date("2023/12/01")
new Date("Fri, 01 Dec 2023 15:43:00 GMT")
new Date("Fri, 01 Dec 2023 20:43:00 EST")
new Date("December 01, 2023 15:43:00")
new Date("1 Dec 2023 15:43:00 GMT")
new Date("2023-12-01 15:43:00")
new Date("2023-12-01T15:43:00")
new Date("2023-12-01 15:43")
```

Reply:
\o/ Looks great and not too loose at the same time.

Only thing that needs to be fixed is to refuse numbers :)

```
> new Date(1701443024549);
2023-12-01T15:03:44.549Z
> new Date(0)
1970-01-01T00:00:00.000Z
> 
```

Reply:
I'm leaning towards not accepting numbers as datetimes. If it's just a year, then the model can probably make sense of it even if it's just number.
If it's an epoch... then hopefully it can guess by the column name and do the right thing. WDYT ?

Reply:
epoch as an int is not that bad no?

Reply:
In a query result, the model can't make any sense of it so it's not as good as an ISO8601 string. However, if the column name suggest it's an epoch ts, then hopefully the model can do the right thing (if we prompt it to) and strftime it